cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaConnectBuild.java

```diff
@@ -139,6 +153,23 @@ private static void validateBuildConfiguration(Build build)    {
         }
     }
 
+    /**
+     * Validates the additional Kaniko options configured by the user against the list of allowed options. If any
+     * options which are not allowed are found, it raises an InvalidResourceException exception.
+     *
+     * @param desiredOptions    List of additional Kaniko options configured by the user
+     */
+    private static void validateAdditionalKanikoOptions(List<String> desiredOptions)    {
+        List<String> allowedOptions = Arrays.asList(DockerOutput.ALLOWED_KANIKO_OPTIONS.split("\\s*,+\\s*"));
+        List<String> forbiddenOptions = desiredOptions.stream()
+                .filter(option -> allowedOptions.stream().noneMatch(option::startsWith))
```

Comment:
`startsWith` risks confusing allowed and disallowed options with differ only by suffix. For example there is already `--insecure` and `--insecure-pull`, but what if `--insecure` was disallowed but `--insecure-pull` was allowed?

I think you've used `startsWith` to avoid having to parse/distinguish from pure `--options` v.s. `--options=with=arguments`? I think either we have the API be a list of pairs with mandatory name and optional argument, or we parse the option name out of the option with optional argument.

Reply:
Ok, I improved the parsing to detect the option name and compare it with equals. Please let me know if that works for you.